File: generate_board.js
import fs from 'fs';
import sharp from 'sharp';
import TextToSVG from 'text-to-svg';

async function generate() {
    const fontPath = './public/fonts/CabinSketch-Bold.ttf';
    const textToSVG = TextToSVG.loadSync(fontPath);

    const text1 = 'The';
    const text2 = 'Dimenstion';

    const attributes = {fill: '#111111', stroke: 'none'};
    const fontSize = 250;
    
    // getMetrics to find width and height
    const m1 = textToSVG.getMetrics(text1, {fontSize});
    const m2 = textToSVG.getMetrics(text2, {fontSize});

    const width = 2048;
    const height = 1024;

    // Center calculations
    const x1 = (width - m1.width) / 2;
    const y1 = height / 2 - m1.height - 20; // top line

    const x2 = (width - m2.width) / 2;
    const y2 = height / 2 + 50; // bottom line

    const path1 = textToSVG.getPath(text1, {x: x1, y: y1, fontSize, anchor: 'top', attributes});
    const path2 = textToSVG.getPath(text2, {x: x2, y: y2, fontSize, anchor: 'top', attributes});

    const compositeSvg = `
        <svg width="${width}" height="${height}" xmlns="http://www.w3.org/2000/svg">
            ${path1}
            ${path2}
        </svg>
    `;

    await sharp('./public/textures/corridor/pustatabliczka.webp')
        .composite([{ input: Buffer.from(compositeSvg), top: 0, left: 0 }])
        .toFile('./public/textures/corridor/readme_board.webp');
        
    console.log('Image generated successfully!');
}

generate().catch(console.error);

File: generate_board.py
from PIL import Image, ImageDraw, ImageFont
import os

# Paths
board_path = "public/textures/corridor/pustatabliczka.webp"
font_path = "public/fonts/CabinSketch-Bold.ttf"
output_path = "public/textures/readme_board.png"

try:
    img = Image.open(board_path).convert("RGBA")
    width, height = img.size
    
    draw = ImageDraw.Draw(img)
    
    font_size_1 = int(height * 0.22)
    font_size_2 = int(height * 0.28)
    
    font1 = ImageFont.truetype(font_path, font_size_1)
    font2 = ImageFont.truetype(font_path, font_size_2)
    
    text1 = "THE"
    text2 = "DIMENSTION"
    
    try:
        bbox1 = draw.textbbox((0, 0), text1, font=font1)
        w1 = bbox1[2] - bbox1[0]
        h1 = bbox1[3] - bbox1[1]
    except AttributeError:
        # Fallback for older PIL
        w1, h1 = draw.textsize(text1, font=font1)
        
    try:
        bbox2 = draw.textbbox((0, 0), text2, font=font2)
        w2 = bbox2[2] - bbox2[0]
        h2 = bbox2[3] - bbox2[1]
    except AttributeError:
        # Fallback for older PIL
        w2, h2 = draw.textsize(text2, font=font2)
    
    x1 = (width - w1) / 2
    y1 = height * 0.2 - (h1 / 2)
    
    x2 = (width - w2) / 2
    y2 = height * 0.55 - (h2 / 2)
    
    draw.text((x1, y1), text1, font=font1, fill=(17, 17, 17, 255))
    draw.text((x2, y2), text2, font=font2, fill=(17, 17, 17, 255))
    
    img.save(output_path, "PNG")
    print("SUCCESS")
except Exception as e:
    print(f"ERROR: {e}")
